src/libstd/rt/libunwind.rs

```diff
@@ -141,3 +146,12 @@ pub unsafe fn _Unwind_RaiseException(exc: *mut _Unwind_Exception)
                                      -> _Unwind_Reason_Code {
     _Unwind_SjLj_RaiseException(exc)
 }
+
+#[cfg(target_os = "nacl")]
+extern "C" {
+    #[link_name = "__pnacl_eh_sjlj_Unwind_RaiseException"]
```

Comment:
I think you should be able to put these attributes on the above definitions with a `#[cfg_attr]` to apply the `link_name`